core/shared/src/main/scala/zio/CancelableFuture.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017-2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio
+
+import scala.concurrent.{ CanAwait, ExecutionContext, Future }
+import scala.concurrent.duration.Duration
+import scala.util.Try
+
+private[zio] abstract class CancelableFuture[+E, +A](val future: Future[A])
+    extends Future[A]
+    with FutureTransformCompat[A] {
+
+  def cancel: Exit[E, A]
```

Comment:
Maybe this could return another `Future` instead of blocking the current thread?

Reply:
I have no strong opinion here, but see
https://github.com/zio/zio/pull/1655#discussion_r323885081

Reply:
Well, if we go back to the original use case:
> We've started introducing zio at the edges of our system. This results in having multiple `unsafeRunToFutures` running in different actors. I'm looking for advice on terminating these from outside of zio (e.g. when the parent actor is shutting down).

I'm fairly sure introducing a potentially never-ending blocking operation inside an akka actor is a big no-no. Note that `Future`/`scala.Promise` is also impure and it's already running the moment it's returned, so it's no less ergonomic to use it in impure code. In fact, I'd expect that most users will not care about the `Exit` return AND might not want to wait until the fiber is interrupted, in which case the default `run-and-forget` behavior of `cancel` without `Await` is preferrable.

Note that there would be no way for a user to achieve `run-and-forget` without blocking a thread if we commit to a synchronous API.

Reply:
👍, Kai. My initial idea was to return an `UIO[Exit[E, A]]`, because the runtime would have been around anyway at the call site where the `CancelableFuture` is created. Returning a `Future` sounds like a good compromise